File: auth-service/src/main/java/com/bank/authservice/security/CurrentUser.java
package com.bank.authservice.security;

import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal;
import java.security.Principal;
import java.util.Collection;
import java.util.Map;
import java.util.UUID;

@Data
@RequiredArgsConstructor
public class CurrentUser implements Principal, OAuth2AuthenticatedPrincipal {
    private final UUID id;
    private final String email;
    private final Map<String, Object> attributes;
    private final Collection<GrantedAuthority> authorities;

    @Override
    public String getName() {
        return email;
    }

    @Override
    public Map<String, Object> getAttributes() {
        return attributes;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return authorities;
    }


}

File: core/src/main/kotlin/ru/patterns/core/security/CurrentUser.kt
package ru.patterns.core.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal
import java.security.Principal
import java.util.UUID

data class CurrentUser(
    val id: UUID,
    private val email: String,
    private val attributes: Map<String, Any>,
    private val authorities: Collection<GrantedAuthority>
) : Principal, OAuth2AuthenticatedPrincipal {
    override fun getName(): String = email
    override fun getAttributes(): Map<String, Any> = attributes
    override fun getAuthorities(): Collection<GrantedAuthority> = authorities

    val fullName: String = attributes["fullName"].toString()

    val isBanned: Boolean = run {
        val enabled = attributes["user_enabled"].toString()
        !enabled.toBoolean()
    }

    val roles: List<String> = authorities
        .map { obj -> obj.authority }
        .map { role -> role.substring(5) }
}